src/executables/master.c

```diff
@@ -527,34 +550,32 @@ css_register_new_server2 (CSS_CONN_ENTRY * conn, unsigned short rid)
 	}
       else
 	{
-	  /* save length info */
-	  server_name_length = name_length;
-
 	  /* accept but make it send us a port id */
 	  css_accept_server_request (conn, SERVER_REQUEST_ACCEPTED_NEW);
-	  name_length = sizeof (buffer);
+	  int name_length = sizeof (buffer);
 	  if (css_net_recv (conn->fd, (char *) &buffer, &name_length, -1) == NO_ERRORS)
 	    {
 #if defined(DEBUG)
 	      css_Active_server_count++;
 #endif
 	      entry =
-		css_add_request_to_socket_queue (conn, false, server_name, conn->fd, READ_WRITE, 0,
+		css_add_request_to_socket_queue (conn, false, server_name.c_str (), conn->fd, READ_WRITE, 0, type,
 						 &css_Master_socket_anchor);
 	      /* store this for later */
 	      if (entry != NULL)
 		{
 		  entry->port_id = ntohl (buffer);
-		  length = (int) strlen (server_name) + 1;
+		  entry->server_type = type;
+		  length = (int) server_name.length () + 1;
```

Comment:
I think this is meant to get the length to the first operator, not the whole string message.
```suggestion
		  length = (int) strlen (server_name.c_str ()) + 1;
```